### PR TITLE
Review Updates

### PR DESCRIPTION
**Pull request by ryo-tagami**
- Add CI:
  - Add `tests/test.yml` and `tests/ansible.cfg` to be able to run
    tests on each commit.
  - Add `tests/Vagrantfile` for local testing.
  - Add `.travis.yml` and `requirements-devel.txt` to run Travis CI on
    every commit.
- Functional Changes:
  - Prefix variables with `midonet_repo_` to avoid variable name
    conflicts.
  - Change the repository file name to `midonet.*` or `midokura.*`
    depending on the flavor(MidoNet vs MEM).
  - Remove `retries` and `becomes` (`becomes` will be applied on
    playbook level).
- Cosmetic and Miscellaneous Changes:
  - Change yum repository name and yum repository file section header
    and apt source list comments, depending on the flavor.
  - Unify string quoting to single-quotes `'`.
  - Add `.gitignore` to prevent garbage from getting committed.
  - Remove unnecessary files.

Signed-off-by: Ryo Tagami rtagami@airstrip.jp

### CODE REVIEW

File: templates/midonet.repo.j2 [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[{{ midonet_repo_repository_section_prefix }}]
+name={{ midonet_repo_repository_description_prefix }}
+baseurl={{ midonet_repo_repository_base_url }}/{{ midonet_repo_midonet_dir }}/{{ midonet_repo_midonet_release|default('stable') }}/el{{ ansible_distribution_major_version }}/
+enabled=1
+gpgcheck=1
+gpgkey={{ midonet_repo_repository_gpgkey_url }}
+
+[{{ midonet_repo_repository_section_prefix }}-openstack-integration]
```
Comment (sibradzic) [inline templates/midonet.repo.j2]: there is no MEM or non-nem openstack repo, but only one thing, name it [midonet-openstack-integration]
same goes for descriptions

File: templates/midonet.repo.j2 [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+name={{ midonet_repo_repository_description_prefix }} OpenStack Integration
+baseurl={{ midonet_repo_repository_base_url }}/openstack-{{ midonet_repo_openstack_plugin_version }}/{{ midonet_repo_openstack_plugin_release|default('stable') }}/el{{ ansible_distribution_major_version }}/
+enabled=1
+gpgcheck=1
+gpgkey={{ midonet_repo_repository_gpgkey_url }}
+
+[{{ midonet_repo_repository_section_prefix }}-misc]
```
Comment (sibradzic) [inline templates/midonet.repo.j2]: there is no MEM or non-nem misc. repo, but only one thing, name it [midonet-misc]
same goes for descriptions

**Review by sibradzic — COMMENTED** [review]
just fix those two descriptions and sections, the rest looks good

File: templates/midonet.repo.j2 (hunk 1 above)
Comment (ryo-tagami) [reply templates/midonet.repo.j2]: Done.

File: templates/midonet.repo.j2 (hunk 2 above)
Comment (ryo-tagami) [reply templates/midonet.repo.j2]: Done.

**Pull request merged**